lib/robots/viomi/capabilities/ViomiMapSegmentEditCapability.js

```diff
@@ -0,0 +1,84 @@
+const MapSegmentEditCapability = require("../../../core/capabilities/MapSegmentEditCapability");
+const ViomiMapParser = require("../ViomiMapParser");
+
+class ViomiMapSegmentEditCapability extends MapSegmentEditCapability {
+    /**
+     *
+     * @param {object} options
+     * @param {import("../../../core/ValetudoRobot")|any} options.robot
+     *
+     * @param {string} options.lang the default language for the generated room names
+     *
+     * @param {object} options.mapActions
+     * @param {number} options.mapActions.JOIN_SEGMENT_TYPE
+     * @param {number} options.mapActions.SPLIT_SEGMENT_TYPE
+     *
+     */
+    constructor(options) {
+        super(options);
+
+        this.lang = options.lang;
+        this.mapActions = options.mapActions;
+    }
+
+    /**
+     * Rounds and formats a point for use in viomi params.
+     *
+     * @param {object} point
+     * @param {number} point.x
+     * @param {number} point.y
+     */
+    pointToViomiString(point) {
+        const roundedX = Math.round((point.x + Number.EPSILON) * 100) / 100;
+        const roundedY = Math.round((point.y + Number.EPSILON) * 100) / 100;
+        return `${roundedX}_${roundedY}`;
+    }
+
+    /**
+     * @param {import("../../../entities/core/ValetudoMapSegment")} segmentA
+     * @param {import("../../../entities/core/ValetudoMapSegment")} segmentB
+     * @returns {Promise<void>}
+     */
+    async joinSegments(segmentA, segmentB) {
+        await this.robot.sendCommand("arrange_room", {
+            lang: this.lang,
+            mapId: this.robot.currentMapId,
```

Comment:
Will this ever be anything that isn't `1`?

Reply:
The currentMapId? For me it currently has a value that looks something like this: `1610520334`

Reply:
Oh that's the unix timestamp for `Wed Jan 13 2021 06:45:34 GMT+0000`

That correlates with the first time you've used the robot I'd assume?

Reply:
Yep that should match up

Reply:
Does it fail if it's not set?

Reply:
Yes, I just tried it and the robot really seems not to like it. The response also takes forever.